streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -128,9 +134,8 @@ public boolean isOpen() {
     @Override
     public synchronized V get(final K key) {
         validateStoreOpen();
-        if (key == null) {
-            return null;
-        }
+        validateKeyNotNull(key);
```

Comment:
Correct - it does. Though i'm wondering if this is what we should do? I'm open to the idea, but i don't really think it is that exceptional so it could be ignored by us. 
Although, it could indicate a bug in user code. 
I guess my preference would be to not throw an exception and just return null